### PR TITLE
CTECH-600 remove java SNAPSHOT

### PR DESCRIPTION
**Pull request by rizwansaeed**
update maven base image

### CODE REVIEW

File: java/publish/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM maven:3.5-jdk-10 as maven
+FROM maven:3.6.3-jdk-11 as maven
```
Comment (MikeMcGarry) [inline java/publish/Dockerfile:1]: Is this necessary for this change?

**Comment by MikeMcGarry** [comment]
Do you know why previously there was a jar-with-dependencies.jar file in the SNAPSHOTS (see screenshot below) but that doesn't seem to exist in the release version (see 2nd screenshot below)?

with-dependencies

no-dependencies

File: java/generate/generate.sh [hunk 2]
```diff
@@ -68,7 +68,7 @@ rm -rf $sdk_output_folder/src/test/java/com/finbourne/lusid/model
 
 # update pom version
 sdk_version=$(cat $swagger_file | jq -r '.info.version')
-mvn -f $sdk_output_folder/pom.xml versions:set -DnewVersion=$sdk_version-SNAPSHOT
+mvn -f $sdk_output_folder/pom.xml versions:set -DnewVersion=$sdk_version
 rm -f $output_folder/.openapi-generator-ignore
```
Comment (MikeMcGarry) [inline java/generate/generate.sh:72]: Do we need to add the gpg plug-in to the POM file for the package to be signed?

```
<build>
  <plugins>
    <plugin>
      <groupId>org.apache.maven.plugins</groupId>
      <artifactId>maven-gpg-plugin</artifactId>
      <version>1.5</version>
      <executions>
        <execution>
          <id>sign-artifacts</id>
          <phase>verify</phase>
          <goals>
            <goal>sign</goal>
          </goals>
        </execution>
      </executions>
    </plugin>
  </plugins>
</build>
```

Comment (rizwansaeed) [reply java/generate/generate.sh:72]: correct, when running the `deploy` task it will sign them

File: java/publish/Dockerfile (hunk 1 above)
Comment (rizwansaeed) [reply java/publish/Dockerfile:1]: the 3.5 (which is an old image) was no longer building due to an old certificate chain and so the `apt-get` were failing

File: java/generate/generate.sh (hunk 2 above)
Comment (rizwansaeed) [reply java/generate/generate.sh:72]: good spot on the dependencies jar, will investigate

Comment (MikeMcGarry) [reply java/generate/generate.sh:72]: Okay great. So on the POM.xml file do we need to add the `maven-gpg-plugin`? Seems like it wasn't necessary even though the docs say that you need it?

Comment (rizwansaeed) [reply java/generate/generate.sh:72]: looks like the dependencies jar got removed with this commit https://github.com/finbourne/lusid-sdk-java-preview/commit/3b20e97acae59a060b4960c3f5190e693e61725b#diff-880de2aacbceb2b255c9e42032e9046bfd764b36bb119e46054a7f76f3049722 in September and the current jars being published don't contain them.

Speaking to Regis, it is accepted practice to not publish the dependencies so given this change is already live, I would say its out of scope for the PR

File: java/generate/config-preview.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+    "developerName": "FINBOURNE Technology Limited",
+    "developerEmail": "engineering@finbourne.com",
+    "developerOrganization": "FINBOURNE Technology Limited",
+    "developerOrganizationUrl": "https://www.finbourne.com",
+    "sourceFolder": "src/main/java",
+    "groupId": "com.finbourne",
+    "artifactId": "lusid-sdk-preview",
```
Comment (rizwansaeed) [inline java/generate/config-preview.json:8]: needs a separate config to name the artifactId appropriately

File: java/generate/templates/pom.mustache [hunk 4]
```diff
@@ -414,12 +441,6 @@
             <version>${commons-lang3-version}</version>
         </dependency>
 
-        <dependency>
```
Comment (rizwansaeed) [inline java/generate/templates/pom.mustache:417]: fix duplicate

File: java/generate/templates/pom.mustache [hunk 5]
```diff
@@ -157,22 +165,7 @@
                         </configuration>
                     </execution>
                 </executions>
-            </plugin>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
```
Comment (rizwansaeed) [inline java/generate/templates/pom.mustache:162]: removing duplicate

File: java/generate/templates/pom.mustache [hunk 6]
```diff
@@ -446,6 +467,7 @@
         <swagger-annotations-version>1.5.21</swagger-annotations-version>
         <jersey-version>1.19.4</jersey-version>
         <jackson-version>2.12.1</jackson-version>
+        <jackson-databind-version>2.12.1</jackson-databind-version>
```
Comment (MikeMcGarry) [inline java/generate/templates/pom.mustache:470]: What is this for?

File: all/generate/build.sh [hunk 7]
```diff
@@ -72,6 +72,12 @@ if [[ -z $lang ]]; then
     
     rm -rf ./output/.git
 
+    # copy custom config to generate folder
```
[the remark this answers is not in the document]
Comment (MikeMcGarry) [reply all/generate/build.sh:75]: Have you test run this with and without a config file?

File: java/generate/templates/pom.mustache (hunk 6 above)
Comment (MikeMcGarry) [reply java/generate/templates/pom.mustache:470]: Nevermind I see that it was above. What was the reference of `${fasterxml-version}` used earlier and now this is hardcoded?

Comment (MikeMcGarry) [reply java/generate/templates/pom.mustache:470]: Oh now I understand, okay new question, where was this version before?

Comment (MikeMcGarry) [reply java/generate/templates/pom.mustache:470]: Ahh I see, you have seperated it from the jackson-version. Okay nice.

File: all/generate/build.sh (hunk 7 above)
[the remark this answers is not in the document]
Comment (MikeMcGarry) [reply all/generate/build.sh:75]: Can you update the README example to include an example of using this too please?

[the remark this answers is not in the document]
Comment (MikeMcGarry) [reply all/generate/build.sh:75]: Thank you.

**Review by MikeMcGarry — APPROVED** [review]

**Pull request merged**